pkg/operator/k8sutil/pod_test.go

```diff
@@ -80,3 +81,30 @@ func TestGetPodPhaseMap(t *testing.T) {
 	// list of failed pods should have 1 entry
 	assert.Equal(t, 1, len(podPhaseMap[v1.PodFailed]))
 }
+
+func TestAddUnreachableNodeToleration(t *testing.T) {
+	podSpec := v1.PodSpec{}
+
+	var tolerationSeconds int64 = 5
+	urToleration := v1.Toleration{Key: "node.kubernetes.io/unreachable",
+		Operator:          "Exists",
+		Effect:            "NoExecute",
+		TolerationSeconds: &tolerationSeconds}
+
+	// Test one toleration of 5 seconds
+	os.Setenv("ROOK_UNREACHABLE_NODE_TOLERATION_LAPSE", "5s")
+	AddUnreachableNodeToleration(&podSpec)
+
+	assert.Equal(t, 1, len(podSpec.Tolerations))
+	assert.Equal(t, urToleration, podSpec.Tolerations[0])
+
+	// Test adding one additional toleration, replaces the previous one,
+	// keepin only the last.
+	os.Setenv("ROOK_UNREACHABLE_NODE_TOLERATION_LAPSE", "6s")
+	tolerationSeconds = 6
+	AddUnreachableNodeToleration(&podSpec)
+
+	assert.Equal(t, 1, len(podSpec.Tolerations))
```

Comment:
This test should also verify that the toleration is appended to the end of a list which doesn't contain the toleration, and it should verify that it alters the toleration when it is found (1) at the beginning, (2) in the middle, and (3) at the end of the toleration list.

Reply:
Done!